controllers/azurecluster_controller.go

```diff
@@ -46,10 +51,28 @@ type AzureClusterReconciler struct {
 }
 
 func (r *AzureClusterReconciler) SetupWithManager(mgr ctrl.Manager, options controller.Options) error {
-	return ctrl.NewControllerManagedBy(mgr).
+	azCluster := &infrav1.AzureCluster{}
+	c, err := ctrl.NewControllerManagedBy(mgr).
 		WithOptions(options).
-		For(&infrav1.AzureCluster{}).
-		Complete(r)
+		For(azCluster).
+		WithEventFilter(predicates.ResourceNotPaused(r.Log)). // don't queue reconcile if resource is paused
+		Build(r)
+	if err != nil {
+		return errors.Wrapf(err, "error creating controller")
+	}
+
+	// Add a watch on clusterv1.Cluster object for unpause & ready notifications.
+	if err = c.Watch(
+		&source.Kind{Type: &clusterv1.Cluster{}},
+		&handler.EnqueueRequestsFromMapFunc{
+			ToRequests: util.ClusterToInfrastructureMapFunc(azCluster.GroupVersionKind()),
+		},
+		predicates.ClusterUnpausedAndInfrastructureReady(r.Log),
```

Comment:
This should just check for unpaused, otherwise we might filter out events that we want to act on, since this controller is responsible for providing the value that populates cluster.status.infrastructureReady
```suggestion
		predicates.ClusterUnpaused(r.Log),
```

Reply:
Changed to unpaused only. TY